### PR TITLE
working isMatchedWith function

### PR DESCRIPTION
**Pull request by TomOverland**

### CODE REVIEW

File: src/matchmaker/matchmaker.js [hunk 1]
```diff
@@ -74,10 +75,59 @@ class Matchmaker {
...
+    this.queue.NW['NW-123456'] = mockUser;
+
+    // Make API call to handlePingResult with client matcher ID = NW-123456
+
+    // get region queue (first 2 chars of client matcherID)
+    const regionQueue = clientMatcherID.split('', 2);
+    const mockRegionQueue = 'NW';
+    console.log('Region Queue: ', regionQueue);
+    // how to tell if player has been claimed as a matcher
+    const opponentMatcherID = this.queue[mockRegionQueue][clientMatcherID]
```
Comment (PeterAlanAnderson) [inline src/matchmaker/matchmaker.js:114]: const opponentMatcherID = this.queue[mockRegionQueue][clientMatcherID]

If you change that to this.queue[regionQueue][clientMatcherID] and drop the hard coded mockRegionQueue does it still work?

File: src/matchmaker/matchmaker.js [hunk 2]
```diff
@@ -74,10 +75,59 @@ class Matchmaker {
...
-      // Empty obj for now, will need to fill in KVP with more info
+      // if port status is "true", it is open.
+      portStatus: true,
     };
     res.json(respObj);
   }
+
+  isMatchedWith(clientMatcherID, res) {
+    // mock opponent values
+    const mockOpponent = {
```
Comment (PeterAlanAnderson) [inline src/matchmaker/matchmaker.js:86]: Can you remove these mock opponents when everything is tested and working?

File: src/matchmaker/matchmaker.js (hunk 1 above)
Comment (TomOverland) [reply src/matchmaker/matchmaker.js]: removed mockRegionQueue

File: src/matchmaker/matchmaker.js (hunk 2 above)
Comment (TomOverland) [reply src/matchmaker/matchmaker.js]: removed and still passes test

**Review by PeterAlanAnderson — APPROVED** [review]

File: src/matchmaker/matchmaker.js [hunk 3]
```diff
@@ -74,10 +75,62 @@ class Matchmaker {
...
+    // console.log('opponentMatcherAddress: ', opponentMatcherAddress);
+    const opponentMatcherPort = this.queue[regionQueue][opponentMatcherID].port;
+    // console.log('opponentMatcherPort: ', opponentMatcherPort);
+    if (typeof opponentMatcherID === 'undefined') {
+      return;
+    }
+
+    const respObj = {
+      shouldStartMatch: true,
+      matchAddress: opponentMatcherAddress,
```
Comment (PeterAlanAnderson) [inline src/matchmaker/matchmaker.js:128]: [DUST]
You could combine line 117 with 128 here:
`matchAddress: this.queue[regionQueue][opponentMatcherID].address;`

The const opponentMatcherAddress isn't used for anything besides being assigned into the object.

**Pull request merged**